### PR TITLE
Added a label option to 'docker-compose run' and test.

### PR DESCRIPTION
**Pull request by smiller123**
Implements #4937.  Designed to work the same way as the -e flag for environment variables.

### CODE REVIEW

**Comment by shin-** [comment]
Thank you! All changes look good to me. cc @albers on the bash completion changes.

File: contrib/completion/bash/docker-compose [hunk 1]
```diff
@@ -403,14 +403,14 @@ _docker_compose_run() {
 			__docker_compose_nospace
 			return
 			;;
-		--entrypoint|--name|--user|-u|--volume|-v|--workdir|-w)
+		--entrypoint|--name|--user|-u|--volume|-v|--workdir|-w|-l)
```
Comment (albers) [inline contrib/completion/bash/docker-compose]: Please insert the option in alphabetical order.

File: contrib/completion/bash/docker-compose [hunk 2]
```diff
@@ -403,14 +403,14 @@ _docker_compose_run() {
...
 			return
 			;;
 	esac
 
 	case "$cur" in
 		-*)
-			COMPREPLY=( $( compgen -W "-d --entrypoint -e --help --name --no-deps --publish -p --rm --service-ports -T --user -u --volume -v --workdir -w" -- "$cur" ) )
+			COMPREPLY=( $( compgen -W "-d --entrypoint -e --help --name --no-deps --publish -p --rm --service-ports -T --user -u --volume -v --workdir -w -l" -- "$cur" ) )
```
Comment (albers) [inline contrib/completion/bash/docker-compose]: Please insert the option in alphabetical order.

**Comment by albers** [comment]
I'd like to have a long form `--label` here as well. Long forms make scripts more readable.

**Comment by smiller123** [comment]
Done! I hadn't had --label at first because it required adding both -l and --label to the usage line for the run command since the docstring parser can't handle repeated options. This is how -e for environment variables work, so I modeled -l for labels off of that.

File: compose/cli/main.py [hunk 3]
```diff
@@ -720,14 +721,17 @@ def run(self, options):
         running. If you do not want to start linked services, use
         `docker-compose run --no-deps SERVICE COMMAND [ARGS...]`.
 
-        Usage: run [options] [-v VOLUME...] [-p PORT...] [-e KEY=VAL...] SERVICE [COMMAND] [ARGS...]
+        Usage:
+            run [options] [-v VOLUME...] [-p PORT...] [-e KEY=VAL...] [-l KEY=VALUE...]
+                [--label KEY=VALUE...] SERVICE [COMMAND] [ARGS...]
```
Comment (shin-) [inline compose/cli/main.py]: You don't need the two versions of the option in the `Usage`, only in the options list below. I would remove the long form for consistency

File: contrib/completion/bash/docker-compose [hunk 4]
```diff
@@ -403,14 +403,14 @@ _docker_compose_run() {
 			__docker_compose_nospace
 			return
 			;;
-		--entrypoint|--name|--user|-u|--volume|-v|--workdir|-w)
+		--entrypoint|-l|--name|--user|-u|--volume|-v|--workdir|-w)
```
Comment (shin-) [inline contrib/completion/bash/docker-compose]: We probably want to add `--label` to the list as well.

File: contrib/completion/bash/docker-compose [hunk 5]
```diff
@@ -403,14 +403,14 @@ _docker_compose_run() {
...
 			return
 			;;
 	esac
 
 	case "$cur" in
 		-*)
-			COMPREPLY=( $( compgen -W "-d --entrypoint -e --help --name --no-deps --publish -p --rm --service-ports -T --user -u --volume -v --workdir -w" -- "$cur" ) )
+			COMPREPLY=( $( compgen -W "-d --entrypoint -e --help -l --name --no-deps --publish -p --rm --service-ports -T --user -u --volume -v --workdir -w" -- "$cur" ) )
```
Comment (shin-) [inline contrib/completion/bash/docker-compose]: ditto

File: tests/acceptance/cli_test.py [hunk 6]
```diff
@@ -1830,6 +1830,20 @@ def test_run_env_values_from_system(self):
         assert 'FOO=bar' in environment
         assert 'BAR=baz' not in environment
 
+    def test_run_label_flag(self):
+        self.base_dir = 'tests/fixtures/run-labels'
+        name = 'service'
+        default = ''
+        foo = 'baz'
+        hello = 'world'
```
Comment (shin-) [inline tests/acceptance/cli_test.py]: These 3 variables are only referenced once in the function - maybe we can substitute the value directly?

File: tests/acceptance/cli_test.py [hunk 7]
```diff
@@ -1830,6 +1830,20 @@ def test_run_env_values_from_system(self):
...
+        self.dispatch(['run', '-l', 'default', '--label', 'foo=baz', name, '/bin/true'])
+        service = self.project.get_service(name)
+        container, = service.containers(stopped=True, one_off=OneOffFilter.only)
+        labels = container.labels
+        self.assertEqual(labels['default'], default)
+        self.assertEqual(labels['foo'], foo)
+        self.assertEqual(labels['hello'], hello)
```
Comment (shin-) [inline tests/acceptance/cli_test.py]: Using `assert A == B` here instead would improve readability.

**Comment by smiller123** [comment]
All done! Thanks!

**Comment by albers** [comment]
Completion LGTM, thanks.

**Pull request merged**